akka-persistence/src/main/scala/akka/persistence/journal/leveldb/LeveldbStore.scala

```diff
@@ -112,6 +112,10 @@ private[persistence] trait LeveldbStore extends Actor with WriteJournalBase with
             batch.delete(keyToBytes(Key(nid, sequenceNr, 0)))
             sequenceNr += 1
           }
+
+          if (this.isInstanceOf[LeveldbCompaction]) {
+            self ! LeveldbCompaction.TryCompactLeveldb(persistenceId, toSeqNr)
+          }
```

Comment:
switching with an `isInstanceOf` here doesn't feel great, but I'm not sure of an alternative... is compaction not applicable to `SharedLeveldbStore`? (then we could mix in `LeveldbCompaction` directly here and skip the check, right?)

Reply:
`LeveldbStore` is also internal api, so you are free to change it, and add mima exclusions to src/main/mima-filters/

Reply:
[SharedLeveldbStore](https://github.com/akka/akka/blob/master/akka-persistence/src/main/scala/akka/persistence/journal/leveldb/SharedLeveldbStore.scala) which extends LeveldbStore though is public, that is why I got the original binary compatibility flag. I assume these mima exclusions allow me to work around this, right?

Reply:
ok, SharedLeveldbStore is not intended for subclassing. It should have been declared final. It's fine to make that exclusion.

Reply:
@patriknw one question before making the commit, given I'm not familiar with mima filters: I have put the filters suggested by the mimaReportBinaryIssues task in a file named _2.5.4.backwards.excludes_ and that seems to do the trick. Would that be an acceptable way to go?

Reply:
yes, that's the way :)